akka-actor/src/main/scala/akka/util/WildcardTree.scala

```diff
@@ -18,15 +18,26 @@ private[akka] final case class WildcardTree[T](data: Option[T] = None, children:
       copy(data = Some(d))
     } else {
       val e = elems.next()
-      copy(children = children.updated(e, children.get(e).getOrElse(WildcardTree()).insert(elems, d)))
+      copy(children = children.updated(e, children.getOrElse(e, WildcardTree()).insert(elems, d)))
     }
 
-  @tailrec final def find(elems: Iterator[String]): WildcardTree[T] =
-    if (!elems.hasNext) this
-    else {
-      (children.get(elems.next()) orElse children.get("*")) match {
-        case Some(branch) ⇒ branch.find(elems)
-        case None         ⇒ WildcardTree()
+  def find(elems: Iterator[String]): WildcardTree[T] = {
+    case class BFSState(wildcardTree: WildcardTree[T], elems: List[String])
+    val queue = mutable.Queue(BFSState(this, elems.toList))
+    while (queue.nonEmpty) {
+      val state = queue.dequeue()
+      if (state.elems.isEmpty) return state.wildcardTree
```

Comment:
no `return`, please (I was staring at the last line `WildcardTree()` and could not understand how this could work ;-))